### PR TITLE
Feature/extend IQLQuery with creation hints as a fallback mechanism

### PR DESCRIPTION
**Pull request by slavistan**

### CODE REVIEW

File: src/attributes/IQLQuery.ts [hunk 1]
```diff
@@ -25,11 +35,25 @@ export class IQLQuery extends AbstractAttributeQuery implements IIQLQuery {
     })
     public queryString: string
 
+    @serialize()
+    @validate({ nullable: true, customValidator: IQLQuery.creationHintsValidator })
+    public creationHints?: {
```
Comment (jkoenig134) [inline src/attributes/IQLQuery.ts:40]: Has to be a class (See relationship Attribute creation hints)

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

File: src/attributes/IQLQuery.ts (hunk 1 above)
Comment (slavistan) [reply src/attributes/IQLQuery.ts:40]: Alright, will change. What's the reason that this needs to be a class?

Comment (jkoenig134) [reply src/attributes/IQLQuery.ts:40]: - safe serialization and validation
- I don't know if serval can handle inline definitions at all
- look at the rest of the codebase

File: src/attributes/IQLQuery.ts [hunk 2]
```diff
@@ -1,14 +1,56 @@
...
+    }
+}
 
 export interface IQLQueryJSON extends AbstractAttributeQueryJSON {
     "@type": "IQLQuery"
     queryString: string
+    creationHints?: {
+        valueType: AttributeValues.Identity.TypeName
+        tags?: string[]
+    }
```
Comment (jkoenig134) [inline src/attributes/IQLQuery.ts:45]:
```suggestion
    creationHints?: IQLQueryCreationHintsJSON
```

File: src/attributes/IQLQuery.ts [hunk 3]
```diff
@@ -1,14 +1,56 @@
...
 }
 
 export interface IIQLQuery extends IAbstractAttributeQuery {
     queryString: string
+    creationHints?: {
+        valueType: AttributeValues.Identity.TypeName
+        tags?: string[]
+    }
```
Comment (jkoenig134) [inline src/attributes/IQLQuery.ts:53]:
```suggestion
    creationHints?: IIQLQueryCreationHints
```

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

**Review by jkoenig134 — APPROVED** [review]

**Review by stnmtz — APPROVED** [review]

**Pull request merged**